Update dev rules to build js files for backend

GD-Dheer opened the pull request.

ds161833 committed:
added a gitignore for node_modules

ds161833 committed:
Merge remote-tracking branch 'origin/develop' into ui-data-fetch

ds161833 committed:
fix

ds161833 committed:
dev env prettify

ds161833 committed:
0

Mehrdadmaskull committed:
Delete .DS_Store

Commit:
Fixing gitignore

Commit:
[#49] Add Dockerfile for mysql

Commit:
[#49] Add mysql Config for docker container

Commit:
[#49] Add & Check initial DB seeding

Commit:
[#49] Fix seeding issue & add persistance to DB changes

Commit:
[#49] Add DB instructions to ReadME

GD-Dheer committed:
Delete .DS_Store

Commit:
Temporary Header

Added blue header to app.js. TEsting out components.

Commit:
[#49] Add configuration for typeORM & connect to DB

Commit:
[#49] wait-for-it script to fix error of backend launching before db

Commit:
Merge branch '49-db-hosting' of https://github.com/Data-Boom/SOEN490 into 49-db-hosting

Commit:
Update dev rules to build js files for backend

Commit:
[#49] Fix typescript checking & Test Dummy DB classes

Commit:
[#49] Additional config for TS

Commit:
Added Splash/Blurb

Added splash/blurb. Used a temporaty image for the splash art.

Commit:
[#49] Verify TypeORM database calling functions

GD-Dheer committed:
Update README.md

GD-Dheer committed:
Update README.md

GD-Dheer committed:
Update README.md

GD-Dheer committed:
Update README.md

ds161833 commented on Contributing.md:
```diff
@@ -28,6 +28,10 @@ Ground Rules for Development
 
     When making commits you should include the issue number in the commit, this allows the commits to be linked the git issue. For example if I was working under my branch '63-Backend-SignUp' and a validator was implemented as part of the feature. The commit would look like '[#63] Add Validator for Input Fields'. 
 
+- For backend, before creating your pull request, please run the following command - it will create the javascript files from the typescript.
```
shouldn't this be automated? part of the npm run start script?

GD-Dheer replied:
I had issues when I tried to integrate with nodemon

ds161833 replied:
i guess lets discuss if we wanna keep TS at all

ds161833 committed:
Merge pull request #60 from Data-Boom/remove-node-modules

Remove node modules

Commit:
Changed fonts, added new splash.

Commit:
[#49] Fix database reinit issue

Commit:
Merge branch '49-db-hosting' of https://github.com/Data-Boom/SOEN490 into 49-db-hosting

Commit:
[#49] Update sql.init to proper dev tables

Commit:
[#49] Update node image & fix indent typo

silent-po2 committed:
[#50] Initial data model and actual data added to database

Commit:
[#49] Update folder directory to include models

Commit:
Merge branch '49-db-hosting' of https://github.com/Data-Boom/SOEN490 into 49-db-hosting

Commit:
Added box with upload button

Added a box where users will go to when they wish to upload data. No functionality to the button so far.

GD-Dheer committed:
Merge pull request #59 from Data-Boom/49-db-hosting

49 db hosting

JeremiahTee committed:
add jest dependency for unit testing

modified package.json to include jest dependency
also added a simple spec.js file as initial dummy test

JeremiahTee committed:
add travis.yml to integrate travis CI

this includes docker service.

added the following docker command:
"docker -compose up -d --build"

JeremiahTee committed:
fix travis yaml file

i tried to trigger a build, it failed because of parsing. the yaml file was invalid -> extra space on Line 5 

http://www.yamllint.com/

JeremiahTee committed:
change node version in yaml file

build failed because the version is too specific and it wasn't found.

Error that I got:

"Version '14.11.1' not found - try `nvm ls-remote` to browse available versions."

JeremiahTee committed:
remove docker compose command 

removing it as travis does not recognize it and it fails the build.

JeremiahTee committed:
correctly install jest as dev dependency (npm test works after this)

build failed again because npm test failed
jest wasn't installed as dev dependency

https://stackoverflow.com/questions/53638220/jest-error-jest-is-not-recognized-as-an-internal-or-external-command-operabl

JeremiahTee committed:
fix "script" section in travis yaml 

this should finally (hopefully) fix the build

JeremiahTee committed:
fix yaml with docker compose, cd Server

added sudo line just in case
added env variable for Server


this yaml file now moves into Server folder, runs npm test, and runs docker compose

lposo committed:
Merge branch 'develop' into ui-import-data

Commit:
Fixed code based on recommendations.

Reorganized Components, Fixed Typos, Changed Blurb to use Box, and added an empty call to api.

Commit:
Merge branch 'ui-import-data' of https://github.com/Data-Boom/SOEN490 into ui-import-data

lposo committed:
Merge pull request #61 from Data-Boom/ui-import-data

UI Import Data

JeremiahTee committed:
reverting commit ff4caa

JeremiahTee committed:
Merge remote-tracking branch 'origin/develop' into develop

JeremiahTee committed:
delete travis.yml file to revert commit

JeremiahTee committed:
delete unneeded tests from previous mocha chai install

JeremiahTee committed:
fix merge conflict package.json

JeremiahTee committed:
fix package.json and include package-lock.json

npm install now properly runs on Server.
all commits are reverted

GD-Dheer committed:
Merge branch 'master' into develop
